File: Interviews/GFG/Daily-Problems/2022/October/Day-18/Print-Diagonally.cpp
class Solution {
public:
	vector<int> downwardDigonal(int N, vector<vector<int>> A) {
		vector<int> ans(N * N);
		for (int i = 0, j = 0, idx = 0; i < 2 * N; i++) {
			if (i == N) continue;
			int k = i < N ? 0 : i % N, l = j;
			while (k < N and l >= 0) {
				ans[idx++] = A[k++][l--];
			}
			j = j == N - 1 ? j : j + 1;
		}

		return ans;
	}
};

File: Interviews/GFG/random-problems/Easy/Print-Diagonally/Print-Diagonally.py
def downwardDigonal(N, A): 
    ans, j, idx = [None for i in range(N * N)], 0, 0
    for i in range(2 * N):
        if i == N: continue
        k, l = 0 if i < N else i % N, j
        while k < N and l >= 0:
            ans[idx] = A[k][l]
            idx, k, l = idx + 1, k + 1, l - 1    
            
        j = j if j == N - 1 else j + 1
    
    return ans